Core/Services/StitchCoreRemoteMongoDBService/Tests/StitchCoreRemoteMongoDBServiceTests/Sync/ChangeEventUnitTests.swift

```diff
@@ -0,0 +1,112 @@
+import Foundation
+import XCTest
+import MongoSwift
+@testable import StitchCoreRemoteMongoDBService
+
+class ChangeEventUnitTests: XCTestCase {
+    private var namespace = MongoNamespace.init(databaseName: ObjectId().description,
+                                                collectionName: ObjectId().description)
+    private let docId = ObjectId()
+    private lazy var document = ["_id": docId,
+                                 "foo": 42,
+                                 "bar": "baz"] as Document
+    private let writePending = false
+
+    func testTransform() {
+        struct TestTransform: Codable {
+            let _id: ObjectId
+            let foo: Int
+            let bar: String
+        }
+        let originalChangeEvent = ChangeEvent<Document>.init(
+            id: Document(),
+            operationType: .insert,
+            fullDocument: ["_id": docId, "foo": 42, "bar": "baz"] as Document,
+            ns: namespace,
+            documentKey: ["_id": docId] as Document,
+            updateDescription: nil,
+            hasUncommittedWrites: writePending)
+        let transformedChangeEvent: ChangeEvent<TestTransform> =
+            ChangeEvent<TestTransform>.transform(changeEvent: originalChangeEvent)
+
+        XCTAssertEqual(originalChangeEvent.id, transformedChangeEvent.id)
+        XCTAssertEqual(originalChangeEvent.operationType, transformedChangeEvent.operationType)
+        XCTAssertEqual(originalChangeEvent.fullDocument?["foo"] as? Int, transformedChangeEvent.fullDocument?.foo)
+        XCTAssertEqual(originalChangeEvent.fullDocument?["bar"] as? String, transformedChangeEvent.fullDocument?.bar)
+        XCTAssertEqual(originalChangeEvent.fullDocument?["_id"] as? ObjectId, transformedChangeEvent.fullDocument?._id)
+        XCTAssertEqual(originalChangeEvent.ns, transformedChangeEvent.ns)
+        XCTAssertEqual(originalChangeEvent.documentKey, transformedChangeEvent.documentKey)
+        XCTAssertNil(transformedChangeEvent.updateDescription)
+        XCTAssertEqual(originalChangeEvent.hasUncommittedWrites, transformedChangeEvent.hasUncommittedWrites)
+    }
+
+    func testChangeEventForLocalInsert() {
+        let expectedLocalInsertEvent = ChangeEvent<Document>.init(
+            id: Document(),
+            operationType: .insert,
+            fullDocument: document,
+            ns: namespace,
+            documentKey: ["_id": docId],
+            updateDescription: nil,
+            hasUncommittedWrites: writePending)
+
+        let actualLocalInsertEvent = ChangeEvent<Document>.changeEventForLocalInsert(namespace: namespace,
+                                                                                     document: document,
+                                                                                     writePending: writePending)
+        XCTAssertEqual(expectedLocalInsertEvent, actualLocalInsertEvent)
```

Comment:
since this uses the the incomplete `==` implementation for change events, this actually isn't comparing all the fields of the expected and actual change event

Reply:
I think this is resolved with the fixed == implementation